src/hooks/useImportExportParams.js

```diff
@@ -12,184 +12,175 @@
 // - a render of a form allowing to modify those values
 // - a function allowing to reset the fields
 
-import React, { useMemo, useState } from 'react';
-import {
-    Autocomplete,
-    Chip,
-    List,
-    ListItem,
-    Switch,
-    Tooltip,
-    Typography,
-    TextField,
-} from '@mui/material';
-import makeStyles from '@mui/styles/makeStyles';
-
-const useStyles = makeStyles((theme) => ({
-    paramListItem: {
-        justifyContent: 'space-between',
-        gap: theme.spacing(2),
-    },
-}));
-
-function longestCommonPrefix(strs) {
-    if (!strs?.length) return '';
-    let prefix = strs.reduce((acc, str) =>
-        str.length < acc.length ? str : acc
-    );
+import React, { useCallback, useMemo, useRef, useState } from 'react';
+import FlatParameters from '../components/FlatParameters/FlatParameters';
+
+const ListRE = /^\[(.*)]$/;
+const sepRE = /[, ]/;
 
-    for (let str of strs) {
-        while (str.slice(0, prefix.length) !== prefix) {
-            prefix = prefix.slice(0, -1);
+function extractDefault(paramDescription) {
+    const d = paramDescription.defaultValue;
+    if (paramDescription.type === 'BOOLEAN') return !!d;
+    if (paramDescription.type === 'DOUBLE') return d - 0.0;
+    if (paramDescription.type === 'INTEGER') return d - 0;
+    if (paramDescription.type === 'STRING_LIST') {
+        if (Array.isArray(d)) return d;
+        const mo = ListRE.exec(d);
+        if (mo?.length > 1) {
+            return mo[1]
+                .split(sepRE)
+                .map((s) => s.trim())
+                .filter((s) => !!s);
         }
+        return [];
     }
-    return prefix;
+    return d ?? null;
+}
+
+function areEquivDeeply(a, b) {
+    if (a === b) return true;
+    const aIsArray = Array.isArray(a);
+    const bIsArray = Array.isArray(b);
+    if (aIsArray || bIsArray) {
+        if (aIsArray && bIsArray && a.length === b.length) {
+            let i = 0;
+            for (; i < a.length && areEquivDeeply(a[i], b[i]); ++i) {}
+            if (i >= a.length) return true;
+        }
+        return false;
+    }
+    const aIsObj = typeof a === 'object';
+    const bIsObj = typeof b === 'object';
+    if (aIsObj || bIsObj) {
```

Comment:
```
if(aIsObj && bIsObj) {
  return areEquivDeeply(Object.entries(a), Object.entries(b));
}
return false;
```
?

Reply:
I wanted to be able to handle corners (like 0, '', null and undefined) cases at the end, but you are right, it can be simplified.

Reply:
Simplified